utils/litemap/src/map.rs

```diff
@@ -221,31 +228,85 @@ impl<K, V> LiteMap<K, V> {
 }
 
 #[cfg(feature = "serde")]
-impl<K: serde::Serialize, V: serde::Serialize> serde::Serialize for LiteMap<K, V> {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        self.values.serialize(serializer)
+mod serde {
+    use serde::{Serialize, Deserialize, Serializer, Deserializer};
+    use serde::ser::SerializeMap;
+    use serde::de::{MapAccess, Visitor};
+    use super::LiteMap;
+    use std::marker::PhantomData;
+    use std::fmt;
+
+    impl<K: Serialize, V: Serialize> Serialize for LiteMap<K, V> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {   
+            let mut map = serializer.serialize_map(Some(self.len()))?;
+                for (k, v) in self.iter() {
+                    map.serialize_entry(k, v)?;
+                }
+            map.end()
+        }
     }
-}
 
-#[cfg(feature = "serde")]
-impl<'de, K: Ord + serde::Deserialize<'de>, V: serde::Deserialize<'de>> serde::Deserialize<'de>
-    for LiteMap<K, V>
-{
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    /// Modified example from https://serde.rs/deserialize-map.html
+    struct LiteMapVisitor<K, V> {
+        marker: PhantomData<fn() -> LiteMap<K, V>>
```

Comment:
I read the example, and I don't really understand why we need LiteMapVisitor to be templatized.  Why can't we `impl<'de, K, V> Visitor<'de> for LiteMapVisitor` ?

Reply:
Then there would be multiple implementations of `Visitor<'de>` for a single type, that's not allowed.